frontend/src/libs/components/article-form/article-form.tsx

```diff
@@ -0,0 +1,103 @@
+import { ButtonType } from '~/libs/enums/button-type.enum.js';
+import { useAppDispatch, useAppForm, useCallback } from '~/libs/hooks/hooks.js';
+import {
+  type ArticleCreateRequestDto,
+  articleCreateValidationSchema,
+} from '~/packages/articles/articles.js';
+import { actions as articlesActions } from '~/slices/articles/articles.js';
+
+import { Button, Input, Textarea } from '../components.js';
+import styles from './styles.module.scss';
+
+const DEFAULT_ARTICLE_FORM_PAYLOAD: ArticleCreateRequestDto = {
+  title: '',
+  text: '',
+  genreId: 1,
+};
+// replace to src/pages/article/components/article-form/libs/constant.ts
+//when Article component will be implemented
+
+const ArticleForm: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { control, errors, handleSubmit, reset } =
+    useAppForm<ArticleCreateRequestDto>({
+      defaultValues: DEFAULT_ARTICLE_FORM_PAYLOAD,
+      validationSchema: articleCreateValidationSchema,
+    });
+
+  const handleArticleSubmit = useCallback(
+    (buttonName: string) =>
+      (payload: ArticleCreateRequestDto): void => {
+        const updatedPayload = {
+          ...payload,
+          ...(buttonName === 'publish' && {
+            publishedAt: new Date().toISOString(),
+          }),
```

Comment:
Let's keep `publishedAt` field consistent in payload

```suggestion
          publishedAt: buttonName === 'publish' ? new Date().toIsoString() : null
```